fulfil_client/client.py

```diff
@@ -189,3 +190,99 @@ def execute(self, records=None, data=None, **kwargs):
                 'context': dumps(context),
             }
         )
+
+
+class AsyncResult(object):
+    """
+    When the server returns an AsyncResult (usually for long running tasks),
+    an instance of this class is created in the response. The object provides
+    a convenient wrapper with which you can poll for the status of the task
+    and result.
+    """
+
+    PENDING = 'PENDING'
+    STARTED = 'STARTED'
+    FAILURE = 'FAILURE'
+    SUCCESS = 'SUCCESS'
+    RETRY = 'RETRY'
+
+    def __init__(self, task_id, token, client):
+        self.task_id = task_id
+        self.token = token
+        self.client = client
+        self.result = None
+        self.state = self.PENDING
+        self.start_time = datetime.utcnow()
+
+    @property
+    def path(self):
+        return '%s/async-result' % (self.client.base_url)
+
+    def bind(self, client):
+        self.client = client
+        return self
+
+    @json_response
+    def _fetch_result(self):
+        if self.client is None:
+            raise Exception(
+                "Unbound AsyncResults cannot refresh status.\n"
+                "Hint: Bind `result.bind(client)` and try again."
+            )
+        return self.client.session.post(
+            self.path,
+            json={
+                'tasks': [
+                    [self.task_id, self.token]
+                ]
+            },
+        )
+
+    def refresh_if_needed(self):
+        """
+        Refresh the status of the task from server if required.
+        """
+        if self.state in (self.PENDING, self.STARTED):
+            response, = self._fetch_result()
```

Comment:
Result will be
```{
'tasks': [<task 1>, ...]
}```

https://github.com/fulfilio/fulfil-server/blob/bb2d4f5d908f077dbe04cfc7cb732a37ee0278ad/trytond_wsgi/api_dispatcher.py#L271